test_scripts/API/VehicleData/OnVehicleData/005_OnVD_App_is_not_subscribed.lua

```diff
@@ -0,0 +1,44 @@
+---------------------------------------------------------------------------------------------------
+-- Description: Check that SDL doesn't transfer OnVehicleData notification with <vd_param> parameter to App
+-- if an app is not subscribed to this parameter
+--
+-- Preconditions:
+-- 1) SDL and HMI are started
+-- 2) SubscribeVehicleData, UnsubscribeVehicleData, OnVehicleData RPCs and
+--  <vd_param> parameter are allowed by policies
+-- 3) App is registered
+-- 4) App is subscribed to <vd_param> parameter
+--
+-- In case:
+-- 1) HMI sends valid OnVehicleData notification with <vd_param> parameter data to SDL
+-- SDL does:
+-- - a) transfer this notification to App
+-- 2) App unsubscribes from <vd_param> parameter
+-- 3) HMI sends valid OnVehicleData notification with <vd_param> parameter data to SDL
+-- SDL does:
+-- - a) not transfer this notification to App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/VehicleData/common')
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment and update preloaded_pt file", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("Register App", common.registerApp)
+
+common.Title("Test")
+for param in common.spairs(common.getVDParams(true)) do
+  common.Title("VD parameter: " .. param)
+  common.Step("RPC " .. common.rpc.sub, common.processSubscriptionRPC, { common.rpc.sub, param })
+  common.Step("RPC " .. common.rpc.on, common.sendOnVehicleData, { param, 1 })
+  common.Step("RPC " .. common.rpc.unsub, common.processSubscriptionRPC, { common.rpc.unsub, param })
+  common.Step("RPC " .. common.rpc.on, common.sendOnVehicleData, { param, 0 })
```

Comment:
maybe this step is redundant, because the notification is also checked in the step below